### ISSUE TITLE
[RMP] Quick-start template pipeline and best practices for training and evaluating retrieval and ranking models

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/Merlin#732]
## Problem:
Merlin platform provides libraries for building models training pipelines, including NVTabular, Dataloader and Merlin Models. When exploring Merlin, customers typically face different learning curves depending on their familiarity with RecSys. Here there is a typical funnel with steps that Data Scientists / ML Engineers get into to assess Merlin and its models with their data:

### New to RecSys
If new to RecSys or implementing their first RecSys pipeline, they need:
1. To perform some basic EDA to better understand the behavior of users in their dataset, in order to choose the most suitable features and models for their context. Generating statistics like the frequency distributions of interactions by users, items, time, sessions help a lot in this process.
2. To find some guideline on whether a single-stage (just retrieval) or two-stage (retrieval + ranking) would be more indicated for their dataset and which models would be suitable for those tasks and available data (e.g. MF x two-tower x session-based models for retrieval).

### New to Merlin:
3. Check our example notebooks and documentation to better understand Merlin features and API
4. Copy-and-paste code snippets from examples and docs and investigate how to glue them together implement a pipeline for their dataset, that allows them to test different models
5. Train a few models and perform some early evaluation, to assess accuracy and performance (runtime / throughput)
6. Setup an offline evaluation with hyperparameter tuning to compare the different models available for the task (e.g. retrieval, ranking) with their data. This requires them to know enough from our API to be able to instantiate correctly the available models, to know how to set the advanced hyperparameters and to find a good search space for hyperparameters optimization. It might be hard for our customers to obtain a reasonable accuracy and performance without too much trial-and-error experimentation on their side.

Merlin provides some examples and documentation for steps 3-5, but it is not straightforward to connect the different available pieces and to adapt them to customers data. And we provide very scarce material for steps 1,2 and 6.

## Goal:
Provide a quick-start (template) pipeline and documented best practices to help customers to quickly build a pipeline that goes from basic EDA for RecSys datasets, to dataset preprocessing with NVTabular to models training and evaluation with Merlin Models.

## Constraints:
...

## Starting Point:
...
